hpy/tools/autogen/__main__.py

```diff
@@ -22,14 +23,28 @@
                     autogen_debug_wrappers,
                     autogen_debug_ctx_call_i)
 from .pypy import autogen_pypy_txt
+from distutils.sysconfig import get_config_var
 
 def main():
     if len(sys.argv) != 2:
         print('Usage: python -m hpy.tools.autogen OUTDIR')
         sys.exit(1)
     outdir = py.path.local(sys.argv[1])
 
-    api = HPyAPI.parse(PUBLIC_API_H)
+    args = get_config_var('CC').split(' ')
+    if sys.platform == 'win32':
+        args += ['/E', '/I' + str(CURRENT_DIR), str(AUTOGEN_H)]
+    else:
+        args += ['-E', '-I' + str(CURRENT_DIR), str(AUTOGEN_H)]
+
+    try:
+        content = subprocess.check_output(args, stderr=subprocess.STDOUT)
+    except subprocess.CalledProcessError as e:
+        print("Preprocessing '%s' failed: " % AUTOGEN_H, file=sys.stderr)
+        print(e.output, file=sys.stderr)
+        raise e
+
+    api = HPyAPI.parse(content.decode('utf-8'))
```

Comment:
I didn't know about the built-in support for the preprocessor. 
If running autogen in Windows is something we want to support, we should probably add a gate for it.
But in this case, the pure Python impl sounds good.

Reply:
I did some investigations and here are my findings:
We cannot directly use `pycparser.parse_file(filename, use_cpp=True)` because we do some manual preprocessing of `public_api.h` with a regex that removes all comments.
Hence, I did best effort and do now use `pycparser.preprocess_file` which is also internally used by `pycparser.parse_file` if `use_cpp=True` is specified.

I did have a look at the `pcpp` package and unfortunately, we cannot use it because it does not support pragmas (see https://github.com/ned14/pcpp#not-implemented-yet-donations-of-code-welcome ) and but that's IMO the only way to annotate C elements.

Concerning the preprocessor problems on Windows that were fixed with #62: I *think* the problems were because we previously just used `use_cpp=True` which, by default, will use the binary `cpp` and AFAIK, there is no such binary in MSVC. My solution uses the C compiler on all platforms and passes `-E` or `/E`. Unfortunately, I cannot test it because I don't have a Windows machine.

Reply:
I'm confused: the cpp preprocessor *does* remove comments. That's why we started to use it in the first place, and the manual regex preprocessing was introduced specifically when we removed the cpp pass.
Anyway, if the current solution works fine on both linux and windows, let's keep it.
Thanks for looking into this!